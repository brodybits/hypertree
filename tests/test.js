import { normalize } from "../src/index.js";

var config = {
  title: "Hello Hyper Tree !",
  nodes: [
    {
      text: "Node1",
      children: [
        {
          text: "Node5"
        }
      ]
    },
    {
      text: "Node2",
      children: [
        {
          text: "Node6",
          children: [
            {
              text: "Node10"
            }
          ]
        },
        {
          text: "Node7"
        }
      ]
    },
    {
      text: "Node3",
      children: [
        {
          text: "Node8"
        },
        {
          text: "Node9",
          children: [
            {
              text: "Node11"
            },
            {
              text: "Node12"
            },
            {
              text: "Node13"
            }
          ]
        }
      ]
    },
    {
      text: "Node4"
    }
  ]
};

const normalizedTree = {
  "ht-node-1": {
    id: "ht-node-1",
    text: "Node1",
    index: 1,
    parentId: "root",
    childrenIds: ["ht-node-5"],
    siblingsIds: ["ht-node-2", "ht-node-3", "ht-node-4"]
  },
  "ht-node-2": {
    id: "ht-node-2",
    text: "Node2",
    index: 2,
    parentId: "root",
    childrenIds: ["ht-node-6", "ht-node-7"],
    siblingsIds: ["ht-node-1", "ht-node-3", "ht-node-4"]
  },
  "ht-node-3": {
    id: "ht-node-3",
    text: "Node3",
    index: 3,
    parentId: "root",
    childrenIds: ["ht-node-8", "ht-node-9"],
    siblingsIds: ["ht-node-1", "ht-node-2", "ht-node-4"]
  },
  "ht-node-4": {
    id: "ht-node-4",
    text: "Node4",
    index: 4,
    parentId: "root",
    childrenIds: [],
    siblingsIds: ["ht-node-1", "ht-node-2", "ht-node-3"]
  },
  "ht-node-5": {
    id: "ht-node-5",
    text: "Node5",
    index: 1,
    parentId: "ht-node-1",
    childrenIds: [],
    siblingsIds: []
  },
  "ht-node-6": {
    id: "ht-node-6",
    text: "Node6",
    index: 1,
    parentId: "ht-node-2",
    childrenIds: ["ht-node-10"],
    siblingsIds: ["ht-node-7"]
  },
  "ht-node-7": {
    id: "ht-node-7",
    text: "Node7",
    index: 2,
    parentId: "ht-node-2",
    childrenIds: [],
    siblingsIds: ["ht-node-6"]
  },
  "ht-node-8": {
    id: "ht-node-8",
    text: "Node8",
    index: 1,
    parentId: "ht-node-3",
    childrenIds: [],
    siblingsIds: ["ht-node-9"]
  },
  "ht-node-9": {
    id: "ht-node-9",
    text: "Node9",
    index: 2,
    parentId: "ht-node-3",
    childrenIds: ["ht-node-11", "ht-node-12", "ht-node-13"],
    siblingsIds: ["ht-node-8"]
  },
  "ht-node-10": {
    id: "ht-node-10",
    text: "Node10",
    index: 1,
    parentId: "ht-node-6",
    childrenIds: [],
    siblingsIds: []
  },
  "ht-node-11": {
    id: "ht-node-11",
    text: "Node11",
    index: 1,
    parentId: "ht-node-9",
    childrenIds: [],
    siblingsIds: ["ht-node-12", "ht-node-13"]
  },
  "ht-node-12": {
    id: "ht-node-12",
    text: "Node12",
    index: 2,
    parentId: "ht-node-9",
    childrenIds: [],
    siblingsIds: ["ht-node-11", "ht-node-13"]
  },
  "ht-node-13": {
    id: "ht-node-13",
    text: "Node13",
    index: 3,
    parentId: "ht-node-9",
    childrenIds: [],
    siblingsIds: ["ht-node-11", "ht-node-12"]
  }
};

test("normalize", () => {
  const result = normalize(config.nodes);

  expect(result).toEqual(normalizedTree);
});
