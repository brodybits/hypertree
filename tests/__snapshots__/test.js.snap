// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view matches snapshots 1`] = `
Object {
  "attributes": Object {},
  "children": Array [
    Object {
      "attributes": Object {},
      "children": Array [
        "Hello Hyper Tree !",
      ],
      "key": null,
      "nodeName": "h1",
    },
    Object {
      "attributes": Object {
        "key": "ht-node-14",
        "onclick": [Function],
        "oncreate": [Function],
        "onremove": [Function],
        "onupdate": [Function],
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            "↑ ",
            "Node1",
          ],
          "key": null,
          "nodeName": "h4",
        },
        Object {
          "attributes": Object {
            "style": Object {
              "display": "block",
              "marginLeft": "10px",
            },
          },
          "children": Array [
            Object {
              "attributes": Object {
                "key": "ht-node-18",
                "onclick": [Function],
                "oncreate": [Function],
                "onremove": [Function],
                "onupdate": [Function],
              },
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [
                    "Node5",
                  ],
                  "key": null,
                  "nodeName": "h4",
                },
                Object {
                  "attributes": Object {
                    "style": Object {
                      "display": "block",
                      "marginLeft": "10px",
                    },
                  },
                  "children": Array [],
                  "key": undefined,
                  "nodeName": "div",
                },
              ],
              "key": "ht-node-18",
              "nodeName": "div",
            },
          ],
          "key": undefined,
          "nodeName": "div",
        },
      ],
      "key": "ht-node-14",
      "nodeName": "div",
    },
    Object {
      "attributes": Object {
        "key": "ht-node-15",
        "onclick": [Function],
        "oncreate": [Function],
        "onremove": [Function],
        "onupdate": [Function],
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            "↑ ",
            "Node2",
          ],
          "key": null,
          "nodeName": "h4",
        },
        Object {
          "attributes": Object {
            "style": Object {
              "display": "block",
              "marginLeft": "10px",
            },
          },
          "children": Array [
            Object {
              "attributes": Object {
                "key": "ht-node-19",
                "onclick": [Function],
                "oncreate": [Function],
                "onremove": [Function],
                "onupdate": [Function],
              },
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [
                    "↑ ",
                    "Node6",
                  ],
                  "key": null,
                  "nodeName": "h4",
                },
                Object {
                  "attributes": Object {
                    "style": Object {
                      "display": "block",
                      "marginLeft": "10px",
                    },
                  },
                  "children": Array [
                    Object {
                      "attributes": Object {
                        "key": "ht-node-23",
                        "onclick": [Function],
                        "oncreate": [Function],
                        "onremove": [Function],
                        "onupdate": [Function],
                      },
                      "children": Array [
                        Object {
                          "attributes": Object {},
                          "children": Array [
                            "Node10",
                          ],
                          "key": null,
                          "nodeName": "h4",
                        },
                        Object {
                          "attributes": Object {
                            "style": Object {
                              "display": "block",
                              "marginLeft": "10px",
                            },
                          },
                          "children": Array [],
                          "key": undefined,
                          "nodeName": "div",
                        },
                      ],
                      "key": "ht-node-23",
                      "nodeName": "div",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "div",
                },
              ],
              "key": "ht-node-19",
              "nodeName": "div",
            },
            Object {
              "attributes": Object {
                "key": "ht-node-20",
                "onclick": [Function],
                "oncreate": [Function],
                "onremove": [Function],
                "onupdate": [Function],
              },
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [
                    "Node7",
                  ],
                  "key": null,
                  "nodeName": "h4",
                },
                Object {
                  "attributes": Object {
                    "style": Object {
                      "display": "block",
                      "marginLeft": "10px",
                    },
                  },
                  "children": Array [],
                  "key": undefined,
                  "nodeName": "div",
                },
              ],
              "key": "ht-node-20",
              "nodeName": "div",
            },
          ],
          "key": undefined,
          "nodeName": "div",
        },
      ],
      "key": "ht-node-15",
      "nodeName": "div",
    },
    Object {
      "attributes": Object {
        "key": "ht-node-16",
        "onclick": [Function],
        "oncreate": [Function],
        "onremove": [Function],
        "onupdate": [Function],
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            "↑ ",
            "Node3",
          ],
          "key": null,
          "nodeName": "h4",
        },
        Object {
          "attributes": Object {
            "style": Object {
              "display": "block",
              "marginLeft": "10px",
            },
          },
          "children": Array [
            Object {
              "attributes": Object {
                "key": "ht-node-21",
                "onclick": [Function],
                "oncreate": [Function],
                "onremove": [Function],
                "onupdate": [Function],
              },
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [
                    "Node8",
                  ],
                  "key": null,
                  "nodeName": "h4",
                },
                Object {
                  "attributes": Object {
                    "style": Object {
                      "display": "block",
                      "marginLeft": "10px",
                    },
                  },
                  "children": Array [],
                  "key": undefined,
                  "nodeName": "div",
                },
              ],
              "key": "ht-node-21",
              "nodeName": "div",
            },
            Object {
              "attributes": Object {
                "key": "ht-node-22",
                "onclick": [Function],
                "oncreate": [Function],
                "onremove": [Function],
                "onupdate": [Function],
              },
              "children": Array [
                Object {
                  "attributes": Object {},
                  "children": Array [
                    "↑ ",
                    "Node9",
                  ],
                  "key": null,
                  "nodeName": "h4",
                },
                Object {
                  "attributes": Object {
                    "style": Object {
                      "display": "block",
                      "marginLeft": "10px",
                    },
                  },
                  "children": Array [
                    Object {
                      "attributes": Object {
                        "key": "ht-node-24",
                        "onclick": [Function],
                        "oncreate": [Function],
                        "onremove": [Function],
                        "onupdate": [Function],
                      },
                      "children": Array [
                        Object {
                          "attributes": Object {},
                          "children": Array [
                            "Node11",
                          ],
                          "key": null,
                          "nodeName": "h4",
                        },
                        Object {
                          "attributes": Object {
                            "style": Object {
                              "display": "block",
                              "marginLeft": "10px",
                            },
                          },
                          "children": Array [],
                          "key": undefined,
                          "nodeName": "div",
                        },
                      ],
                      "key": "ht-node-24",
                      "nodeName": "div",
                    },
                    Object {
                      "attributes": Object {
                        "key": "ht-node-25",
                        "onclick": [Function],
                        "oncreate": [Function],
                        "onremove": [Function],
                        "onupdate": [Function],
                      },
                      "children": Array [
                        Object {
                          "attributes": Object {},
                          "children": Array [
                            "Node12",
                          ],
                          "key": null,
                          "nodeName": "h4",
                        },
                        Object {
                          "attributes": Object {
                            "style": Object {
                              "display": "block",
                              "marginLeft": "10px",
                            },
                          },
                          "children": Array [],
                          "key": undefined,
                          "nodeName": "div",
                        },
                      ],
                      "key": "ht-node-25",
                      "nodeName": "div",
                    },
                    Object {
                      "attributes": Object {
                        "key": "ht-node-26",
                        "onclick": [Function],
                        "oncreate": [Function],
                        "onremove": [Function],
                        "onupdate": [Function],
                      },
                      "children": Array [
                        Object {
                          "attributes": Object {},
                          "children": Array [
                            "Node13",
                          ],
                          "key": null,
                          "nodeName": "h4",
                        },
                        Object {
                          "attributes": Object {
                            "style": Object {
                              "display": "block",
                              "marginLeft": "10px",
                            },
                          },
                          "children": Array [],
                          "key": undefined,
                          "nodeName": "div",
                        },
                      ],
                      "key": "ht-node-26",
                      "nodeName": "div",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "div",
                },
              ],
              "key": "ht-node-22",
              "nodeName": "div",
            },
          ],
          "key": undefined,
          "nodeName": "div",
        },
      ],
      "key": "ht-node-16",
      "nodeName": "div",
    },
    Object {
      "attributes": Object {
        "key": "ht-node-17",
        "onclick": [Function],
        "oncreate": [Function],
        "onremove": [Function],
        "onupdate": [Function],
      },
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            "Node4",
          ],
          "key": null,
          "nodeName": "h4",
        },
        Object {
          "attributes": Object {
            "style": Object {
              "display": "block",
              "marginLeft": "10px",
            },
          },
          "children": Array [],
          "key": undefined,
          "nodeName": "div",
        },
      ],
      "key": "ht-node-17",
      "nodeName": "div",
    },
  ],
  "key": null,
  "nodeName": "div",
}
`;
